packages/react-components/src/utils.ts

```diff
@@ -31,6 +31,9 @@ export const isInternalLink = (href: string): boolean =>
   globalThis.location &&
   new URL(href, globalThis.location.href).origin === globalThis.location.origin;
 
+export const isHashLink = (href: string): boolean =>
+  globalThis.location && !!new URL(href, globalThis.location.href).hash;
```

Comment:
I could see this being a problem with `/#`, where `.hash` is `''`

Reply:
as evidenced by one of the tests

Reply:
They both scroll to the top of the page

Reply:
The non-hashlink because of the custom code? :/